pkg/apis/triggers/v1alpha1/cluster_interceptor_types.go

```diff
@@ -0,0 +1,107 @@
+package v1alpha1
+
+import (
+	"errors"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+// Check that EventListener may be validated and defaulted.
+var _ apis.Validatable = (*ClusterInterceptor)(nil)
+var _ apis.Defaultable = (*ClusterInterceptor)(nil)
+
+// +genclient
+// +genclient:nonNamespaced
+// +genreconciler:krshapedlogic=false
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+// ClusterInterceptor describes a pluggable interceptor including configuration
+// such as the fields it accepts and its deployment address. The type is based on
+// the Validating/MutatingWebhookConfiguration types for configuring AdmissionWebhooks
+type ClusterInterceptor struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ClusterInterceptorSpec `json:"spec"`
+	// +optional
+	Status ClusterInterceptorStatus `json:"status"`
+}
+
+// ClusterInterceptorSpec describes the Spec for an ClusterInterceptor
+type ClusterInterceptorSpec struct {
+	ClientConfig ClientConfig `json:"clientConfig"`
+}
+
+// ClusterInterceptorStatus holds the status of the ClusterInterceptor
+// +k8s:deepcopy-gen=true
+type ClusterInterceptorStatus struct {
+	duckv1.Status `json:",inline"`
+
+	// ClusterInterceptor is Addressable and exposes the URL where the Interceptor is running
+	duckv1.AddressStatus `json:",inline"`
+}
+
+// ClientConfig describes how a client can communicate with the Interceptor
+type ClientConfig struct {
+	// URL is a fully formed URL pointing to the interceptor
+	// Mutually exclusive with Service
+	URL *apis.URL `json:"url,omitempty"`
+
+	// Service is a reference to a Service object where the interceptor is running
+	// Mutually exclusive with URL
+	Service *ServiceReference `json:"service,omitempty"`
+}
+
+// ServServiceReference is a reference to a Service object
+// with an optional path
+type ServiceReference struct {
+	// Name is the name of the service
+	Name string `json:"name"`
+
+	// Namespace is the namespace of the service
+	Namespace string `json:"namespace"`
+
+	// Path is an optional URL path
+	// +optional
+	Path string `json:"path"`
```

Comment:
can we add `omitempty` as its optional field

Reply:
done